### DISCUSSION TITLE
Setup & workflow for contributions

### DISCUSSION THREAD

**Discussion by ybakos** [osu-cascades/ecotone-web#190] [General]
@LexiSchindel let's bring the discussion from https://github.com/osu-cascades/ecotone-web/pull/183 to here.

**Comment by ybakos** [comment]
Do you, in your local repo, have a remote named "upstream" that points to https://github.com/osu-cascades/ecotone-web ?

Shout back if I am not making sense.

**Reply by LexiSchindel** [reply]
Yes, I did have upstream set to look at ecotone-web.

**Comment by LexiSchindel** [comment]
Apologies for the delay! I pulled in the upstream changes to my main branch. Just opened a new PR. Let me know if this one ends up working. 

If not, I'll delete my fork and redo the changes without setting up the environment to hopefully get this resolved.

**Comment by ybakos** [comment]
Thanks for taking the time to give it another spin! We will get these changes in but, and please don't hate me, what we need is a "clean" PR that just has the commits and changes to README.md and CONTRIBUTING.

To get there, you don't need to delete and re-fork*. I _think_ you would (correct me / re-interpret here):

1. Ensure the local master and develop branches are up to date with upstream: `git checkout master; git pull upstream master; git checkout develop; git pull upstream develop`
2. Create a local feature branch, off of develop: `git checkout develop; git checkout -b improve_contributing`
3. Apply your changes to README.md and CONTRIBUTING, and commit. (You can [cherry pick](https://git-scm.com/docs/git-cherry-pick) here or just manually apply the changes with some clever copy/paste.
4. Push your feature branch `git push origin improve_contributing`
5. Create a new PR from _improve_contributing_ to _develop_.

* If you made commits on your local master or develop branch, you can try using `git reset` to remove your local commits. Or, if simpler, you can indeed delete your local repo and re-fork.

**Comment by LexiSchindel** [comment]
Okay, another try is ready! Hopefully this one does the trick. I added details on how to set upstream, branch, and pull to the contributing guidelines as well. 

https://github.com/osu-cascades/ecotone-web/pull/193
